src/NEPTypes.jl

```diff
@@ -102,19 +111,31 @@ All of the matrices ``A_0,...`` are of size ``n×n``
 and ``f_i`` are a functions. The  functions ``f_i`` must be defined
 for matrices in the standard matrix function sense.
 """
-    struct SPMF_NEP <: AbstractSPMF
+
+# Logic behind Ftype:
+#  The eltype(F(λ))=promote_type(eltype(λ),Ftype)
+
+    struct SPMF_NEP_dense{T<:AbstractMatrix,Ftype}  <: AbstractSPMF{T}  # T should be a AbstractMatrix
+        n::Int
+        A::Vector{T}   # Array of Array of matrices
+        fi::Vector{Function}  # Array of functions
+        Schur_factorize_before::Bool # Tells if you want to do the Schur-factorization
+    end
+
+    struct SPMF_NEP_sparse{T,Ftype}  <: AbstractSPMF{T}  # T should be a AbstractMatrix
```

Comment:
Here we could also do `T<:AbstractMatrix`, right?

Reply:
Yep.